solutions/eskimo/assignment1/command_line.js

```
@@ -0,0 +1,341 @@
+                var command_exist=0;
+
+                //Checks commandList for a match
+                this.commandList.forEach((command_available)=>{
+
+                    //Checks if argument starts with a command specified
+                    if(val.startsWith(command_available.command))
+                    {
+                        //Sets command_exists true
+                        command_exist=1;
+                        
+                        //Checks if this command have a exclusive property
+                        if(command_available.ExclusiveIndex!=null)
+                        {
+                            //Checks if this command has previously occured or not 
+                            if(this.isSetExclusive[command_available.ExclusiveIndex]!="0")
+                            {
+                                //Both commands can not occur together
+                                console.log("The "+command_available.command+" and "+this.isSetExclusive[command_available.ExclusiveIndex]+" arguments cannot be used together.")
+                                return;
+                            }
+                            else
+                            {
+                                //Sets index EclusiveIndex of isSetExclusive with command name
+                                this.isSetExclusive[command_available.ExclusiveIndex]=command_available.command;
+                            }
+                        }
+
+                        //Checks for command demandOption
+                        if(command_available.demandOption)
+                        {
+                            //Checks for '=' in argument if demandOption is true
+                            if(!val.includes("="))
+                            {
+                                //Error if argument doesn't have a value
+                                console.log("The '--"+command_available.command+"' argument is required, but missing from input.");
```

Instead of logging, you should throw errors here too, right?